File: auth-and-payments/rails/test/services/composition_test.rb
require "test_helper"
class CompositionTest < ActiveSupport::TestCase
  test "auth and payments share one validated pairwise subject" do
    auth=VibeNestAuth.new(issuer:"https://vibenest.net/",client_id:"client",client_secret:nil,redirect_uri:"https://reference.example/auth/vibenest/callback"); flow=auth.begin_flow("issuer"=>"https://vibenest.net/","authorization_endpoint"=>"https://vibenest.net/oauth/authorize"); assert flow[:pending]["nonce"].present?
    path=Rails.root.join("tmp","combined-#{SecureRandom.hex(8)}.sqlite").to_s
    payments=nil
    begin payments=ProjectPayments.new(secret:"s"*32,environment_id:"sim_environment_reference",database_path:path); checkout=payments.checkout("validated-pairwise-sub",{"priceKey"=>"monthly"}); assert_not_includes checkout[:url],"validated-pairwise-sub"; assert_includes checkout[:url],ProjectPayments.customer_id("validated-pairwise-sub")
    ensure payments&.close; File.delete(path) if File.exist?(path); end
  end
end

File: auth-and-payments/laravel/tests/CompositionTest.php
<?php
namespace Tests;
use App\Services\ProjectPayments; use App\Services\VibeNestAuth; use PHPUnit\Framework\TestCase;
final class CompositionTest extends TestCase
{
 public function testAuthAndPaymentsShareTheSamePairwiseSubject():void{$auth=new VibeNestAuth('https://vibenest.net/','client',null,'https://reference.example/auth/vibenest/callback');$flow=$auth->begin(['issuer'=>'https://vibenest.net/','authorization_endpoint'=>'https://vibenest.net/oauth/authorize']);self::assertArrayHasKey('nonce',$flow['pending']);$file=sys_get_temp_dir().'/vn-combined-'.bin2hex(random_bytes(8)).'.sqlite';try{$payments=new ProjectPayments(str_repeat('s',32),'sim_environment_reference',$file);$checkout=$payments->checkout('validated-pairwise-sub',['priceKey'=>'monthly']);self::assertStringNotContainsString('validated-pairwise-sub',$checkout['url']);self::assertStringContainsString(ProjectPayments::customerId('validated-pairwise-sub'),$checkout['url']);}finally{@unlink($file);}}
}
